pkg/netid/netid.go

```diff
@@ -0,0 +1,61 @@
+package netid
+
+/*
+Accessor methods to annotate namespace for multitenant support
+*/
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+const (
+	// Current assigned VNID for the namespace
+	VNIDAnnotation string = "pod.network.openshift.io/multitenant.vnid"
+	// Desired VNID for the namespace
+	WantsVNIDAnnotation string = "pod.network.openshift.io/multitenant.wants-vnid"
```

Comment:
Currently we do *not* allow unprivileged admins to request VNID.
The use cases addressed by vnid allocator is slightly different from other existing allocators.
(1) Assign/Release VNID during project creation/deletion respectively:
     vnid controller performs this task (similar to other allocators)
(2) Allow admin to manipulate VNIDs using oadm cmd:
     Some vnids are shared by multiple namespaces and release is done only when no namespace needs that vnid.
     If we allow admin cli to directly manipulate the vnid annotation. We could end up in a situation where admin CLI patches vnid in a namespace but the same vnid could be released by controller at the same time. We can't provide synchronization here because only vnid controller knows which ids are allocated. wants-vnid annotation provides this synchronization. Admin CLI requests needed vnid as wants-vnid annotation and vnid controller verifies if the requested vnid is allowed or not and updates vnid annotation. wants-vnid annotation is removed after every request (either success or failure). Eventually the idea is to replace wants-vnid annotation with network-policy objects (once upstream implements https://github.com/kubernetes/kubernetes/pull/24154).